src/game_event.cpp

```diff
@@ -412,26 +412,30 @@ void Game_Event::UpdateSelfMovement() {
 }
 
 void Game_Event::MoveTypeRandom() {
+	SetMaxStopCountForStep();
 	int last_direction = GetDirection();
-	switch (Utils::GetRandomNumber(0, 5)) {
-	case 0:
-		SetStopCount(GetStopCount() - Utils::GetRandomNumber(0, GetStopCount()));
-		if (GetStopCount() < 0) {
-			SetStopCount(0);
-		}
+
+	int draw = Utils::GetRandomNumber(0, 9);
+
+	const auto opt = MoveOption::IgnoreIfCantMove;
+
+	if (draw < 3) {
+		auto dir = GetDirection();
+		Move(dir, opt);
+	} else if (draw < 5) {
+		auto dir = GetDirection90DegreeLeft(GetDirection());
+		Move(dir, opt);
+	} else if (draw < 7) {
+		auto dir = GetDirection90DegreeRight(GetDirection());
+		Move(dir, opt);
+	} else if (draw < 8) {
+		auto dir = GetDirection180Degree(GetDirection());
+		Move(dir, opt);
 		return;
-	case 1:
-		MoveForward();
-		break;
-	default:
-		MoveRandom();
-	}
-	if (move_failed && !starting) {
-		SetDirection(last_direction);
-		if (!(IsDirectionFixed() || IsFacingLocked()))
-			SetSpriteDirection(last_direction);
 	} else {
-		SetMaxStopCount(GetMaxStopCount() / 5 * Utils::GetRandomNumber(3, 6));
+		// As per Cherry, RPG_RT supposedly waits for rand(0.. max_stop_count) inclusive in this case.
+		// This has not been observed empirically, as save data shows it only waiting 1 frame.
+		return;
```

Comment:
guess this is a good use case for dynrpg, will see if I can get this somehow dumped.

Reply:
*I believe* that `max_stop_count` is the incorrectly named `RPG::Character::movementFramesLeft` and `stop_count` is (also not quite correctly named) `RPG::Character::stepFrameCounter`.

Reply:
Looking at dumps I believe `RPG::Character::movementFramesLeft` is our `remaining_step`